iroh/src/sync_engine/rpc.rs

```diff
@@ -180,6 +180,7 @@ impl<S: Store> SyncEngine<S> {
         let tag = bao_store
             .import_bytes(value.into(), BlobFormat::RAW)
             .await?;
+        // TODO: Use sync handle
```

Comment:
alle the store operations probably should? that way there is no need anymore for the spawn_blocking calls either

Reply:
Yes, that's the plan I think. Will do.

Reply:
Actually, I'm not that sure anymore: Redb is basically a global rwlock, correct?
By passing everything through a single channel onto a single thread, we basically make that a mutex. So no concurrent reads can happen on the database. Whereas before, with transforming the sync interators into streams with `spawn_blocking`, this could at least theoretically happen in parallel if no thread is writing at the time. Depending on how many threads the blocking pool has.

So do we want a threadpool of our own here in the end?

Maybe these thoughts are premature, and it makes sense architecture wise to have all interactions go through the channel interface, to have a clean point between sync and async interactions. And we could then still add a threadpool for non-mutating access to the replica later...

Reply:
my initial sense would be for now to make everything go to the actor, and potentially parallelise in there if possible later down the road